src/blueapi/service/model.py

```diff
@@ -174,6 +174,25 @@ class EnvironmentResponse(BlueapiBaseModel):
     )
 
 
+class RepositoryStatus(BlueapiBaseModel):
+    remote_url: str = Field(description="URL of a repository on a remote server")
+    version: str = Field(description="Branch or tag of the repository")
+    is_dirty: bool = Field(description="Does the repository have uncommitted changes")
+
+
+class ScratchResponse(BlueapiBaseModel):
+    """
+    State of the scratch area.
+    """
+
+    package_info: list[RepositoryStatus] = Field(
```

Comment:
Nit: From the name this sounds like a single object rather than a list of objects

```suggestion
    packages: list[RepositoryStatus] = Field(
```